ne_lint/yamllint_ext/rules/node_templates.py

```diff
@@ -572,6 +572,45 @@ def check_supports_tagging(model, line):
                 'parameter in properties. A best practice is to provide Tags.'
                 'For example: https://tinyurl.com/yveu36xs'
                 .format(node=model.name, type=model.node_type))
+        else:
+            yield from validate_tags_property_struct(model, line)
+
+
+def validate_tags_property_struct(model_, line_):
+
+    tags = model_.properties.get('Tags')
+    if not isinstance(tags, list):
+        yield LintProblem(
+            line_,
+            None,
+            'The node template {node} with {type} does not provide a valid'
+            ' structure of the Tags property. Please ensure that each '
+            'the Tags property is a list. '
+            'For example: https://tinyurl.com/yveu36xs'
+            .format(node=model_.name, type=model_.node_type))
+        return
+
+    for tag in tags:
+        if not isinstance(tag, dict):
+            yield LintProblem(
+                line_,
+                None,
+                'The node template {node} with {type} does not provide a valid'
+                ' structure of the Tags property. Please ensure that each '
+                'tag is a dict. '
+                'For example: https://tinyurl.com/yveu36xs'
+                .format(node=model_.name, type=model_.node_type))
+            return
+        if set(tag.keys()) != {'Key', 'Value'}:
```

Comment:
That's a nice way to do it. I like it.
Another way would be:
```suggestion
        if list(tag.keys()) != ['Key', 'Value']):
```
And it might be a little faster:
```
def validate_list(validate_me):
    return list(validate_me.keys()) != ['foo', 'baz']
def validate_set(validate_me):
    return set(validate_me.keys()) != {'foo', 'baz'}
def check_time():
    t = time.time()
    print(validate_set(d))
    n = time.time()
    a = n - t
    t = time.time()
    print(validate_list(d))
    n = time.time()
    b = n - t
    print(f'Set is {b - a} faster than list')
```
Just how much faster depends on other circumstances, but look at these results:
```
In [18]: check_time()
False
False
Set is -0.00015974044799804688 faster than list

In [19]: check_time()
False
False
Set is -6.937980651855469e-05 faster than list

In [20]: check_time()
False
False
Set is -0.00018215179443359375 faster than list

In [21]: check_time()
False
False
Set is -8.821487426757812e-05 faster than list```

Reply:
Thats a great observation, thanks, I changed it. When uniqueness is not an issue and set operations (like union, intersection, difference) is not needed, its redundant to use set as it creates hash code for each key.